### PR TITLE
#1130 - Add constraints list of zonotope test

### LINKED ISSUE #1130: Constraints list of zonotope dim > 2 contains 'undef'
```julia
julia> Z = rand(Zonotope, dim=3)
Zonotope{Float64}([0.986558, -0.597589, 2.12338], [0.827946 1.425 … 2.23784 1.19382; -0.0368336 -0.561346 … 0.527 1.34382; 0.125808 0.716952 …
 -0.903953 -2.19089])

julia> constraints_list(Z)

20-element Array{HalfSpace{Float64},1}:
    HalfSpace{Float64}([0.0754287, -0.706835, -0.703345], 1.4691567127403207)
    HalfSpace{Float64}([-0.155973, -0.207414, 0.965739], 6.51935266681345)
    HalfSpace{Float64}([-0.0286078, 0.892749, 0.449646], 2.3527903180709413)
    HalfSpace{Float64}([-0.0387381, 0.861055, 0.507035], 2.319388716483151)
    HalfSpace{Float64}([-0.527798, -0.353352, 0.772381], 6.519994229835396)
    HalfSpace{Float64}([0.0367844, 0.821019, 0.569714], 2.3505707559324427)
    HalfSpace{Float64}([0.0560637, 0.837179, 0.544048], 2.2972784381091524)
    HalfSpace{Float64}([-0.366035, -0.0404095, -0.929723], 0.8185430286146076)
   ⋮
    HalfSpace{Float64}([0.366035, 0.0404095, 0.929723], 5.440781406186192)
    HalfSpace{Float64}([0.764019, 0.273766, 0.584233], 5.206505925973481)
    HalfSpace{Float64}([-0.0133573, -0.849089, -0.528082], 0.973836297297932)
 #undef
 #undef
 #undef
 #undef
```

### PR DESCRIPTION
**Pull request by mforets**
Closes #1130.

### CODE REVIEW

File: test/unit_Zonotope.jl [hunk 1]
```diff
@@ -101,6 +101,18 @@ for N in [Float64, Rational{Int}, Float32]
     Z = Zonotope(N[0, 0], N[1 1; -1 1])
     Z1, Z2 = split(Z, 1) # in this case the splitting is exact
     @test Z1 ⊆ Z && Z2 ⊆ Z
+
+    # list of constraints
+    Z = Zonotope(zeros(N, 3), Matrix(N(1)*I, 3, 3))
+    B = BallInf(zeros(N, 3), N(1)) # equivalent to Z
+    if N == Rational{Int} && !test_suite_polyhedra
+        # the rational cases uses vrep => needs Polyhedra
+        nothing
```
Comment (schillic) [inline test/unit_Zonotope.jl]: Should we not rather write `if N != Rational{Int} || test_suite_polyhedra` and omit the `nothing` case?

File: test/unit_Zonotope.jl [hunk 2]
```diff
@@ -101,6 +101,18 @@ for N in [Float64, Rational{Int}, Float32]
     Z = Zonotope(N[0, 0], N[1 1; -1 1])
     Z1, Z2 = split(Z, 1) # in this case the splitting is exact
     @test Z1 ⊆ Z && Z2 ⊆ Z
+
+    # list of constraints
+    Z = Zonotope(zeros(N, 3), Matrix(N(1)*I, 3, 3))
+    B = BallInf(zeros(N, 3), N(1)) # equivalent to Z
+    if N == Rational{Int} && !test_suite_polyhedra
+        # the rational cases uses vrep => needs Polyhedra
```
Comment (schillic) [inline test/unit_Zonotope.jl]:
```suggestion
        # the rational case uses vrep => needs Polyhedra
```

File: test/unit_Zonotope.jl (hunk 1 above)
Comment (mforets) [reply test/unit_Zonotope.jl]: yes, they are logically equivalent :) i was thinking "if this happens => ommit this test".  i can make the suggested change and invert the condition.

**Comment by mforets** [comment]
Thanks for the review 👍

**Pull request merged**